sim/battle-actions.ts

```diff
@@ -1054,6 +1054,7 @@ export class BattleActions {
 		const pokemonOriginalHP = pokemon.hp;
 		if (damagedDamage.length && !isSecondary && !isSelf) {
 			this.battle.runEvent('DamagingHit', damagedTargets, pokemon, move, damagedDamage);
+			this.battle.runEvent('TakeDamage', damagedTargets, pokemon, null, damagedDamage);
```

Comment:
Might as well put this line below the AfterHit part so you can remove the hardcode in Knock Off.